ndk/src/media/media_codec.rs

```diff
@@ -510,3 +529,11 @@ impl OutputBuffer<'_> {
         self.info.presentationTimeUs
     }
 }
+
+#[derive(Debug)]
+pub enum DequeuedOutputBufferResult<'a> {
```

Comment:
I wonder if this should be named more like the underlying NDK type `DequeuedOutputBufferInfo`.

In the C api the same info is also used in a few other places, namely [AMediaMuxer_writeSampleData](https://developer.android.com/ndk/reference/group/media#group___media_1gab516a49c6cb40440efe33fa2237b7e1b) and https://developer.android.com/ndk/reference/group/media#amediacodeconasyncoutputavailable so maybe it's worth double checking if it also makes sense to have a general type in the Rust API too.

`Info` feels like it also helps distinguish it conceptually from the `Result()` that it's returned within.

Reply:
I suppose the issue with that suggestion though is that this version isn't really just 'info' it may also encapsulate a buffer.

Reply:
`Info` also suits my fancy, wasn't really a fan of `Result` already: https://github.com/rust-mobile/ndk/pull/316#issuecomment-1594340070

Reply:
> In the C api the same info is also used in a few other places, namely [AMediaMuxer_writeSampleData](https://developer.android.com/ndk/reference/group/media#group___media_1gab516a49c6cb40440efe33fa2237b7e1b) and https://developer.android.com/ndk/reference/group/media#amediacodeconasyncoutputavailable so maybe it's worth double checking if it also makes sense to have a general type in the Rust API too.

Fair point, guess we should make a separate PR to generalize this type so that `AMediaMuxer` can be implemented at some point. CC @zarik5.

Reply:
Or how about `*BufferInfoResult`?

Reply:
hehe, yeah always secret option number three to combine them :)

no strong preference from me really. Feels pretty verbose but yeah maybe BufferInfoResult helps connect to the underlying NDK API while also representing that it's not pure info in this case.

Reply:
Let's go for it, I think this was enough bikeshedding. Leaving `DequeuedInputBufferResult` as it is because there's no `Info` struct involved.

Generalizing `DequeuedOutputBufferInfoResult` is TBD, the inner `OutputBuffer` struct is quite specific to `MediaCodec`.